dashboard-api/internal/account.go

```diff
@@ -0,0 +1,85 @@
+package studiesmanager
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"time"
+)
+
+type AccountRepository interface {
+	Create(ctx context.Context, a Account) error
+}
+
+var ErrAccountAlreadyExists = errors.New("Account Already Exists")
+
+//go:generate mockery --case=snake --outpkg=storagemocks
+//--output=platform/storage/storagemocks --name=AccountRepository
+
+type Account interface {
+	GetName() string
+	GetUserID() string
+	GetAuthType() string
+	MarshalCredentials() (string, error)
+}
+
+type AccountBase struct {
+	ID        string    `json:"id" `
+	UserID    string    `json:"userId"`
+	AuthType  string    `json:"authType" validate:"required"`
+	Name      string    `json:"name" validate:"required"`
+	CreatedAt time.Time `json:"createdAt"`
+}
+
+func (a AccountBase) GetName() string {
+	return a.Name
+}
+
+func (a AccountBase) GetUserID() string {
+	return a.UserID
+}
+
+func (a AccountBase) GetAuthType() string {
+	return a.AuthType
+}
+
+type FlyAccount struct {
+	AccountBase
+	Credentials FlyCredentials `json:"credentials" validate:"required"`
+}
+
+// TODO: we could use the type in `inference/sources/fly/main.go`
+// for this, however we would need to add the validator to it
+// this also might break "domain" interfaces and be hard to decouple later
+type FlyCredentials struct {
+	APIKey string `json:"api_key" validate:"required"`
+}
```

Comment:
@nandanrao let me know your thoughts on this

Reply:
Adding validator tags seems harmless to me!

Reply:
For reference, I started this work and then came across that that source is actually a executable and not a package so we need to refactor it:
```
internal/account.go:9:2: import "github.com/vlab-research/vlab/inference/sources/fly" is a program, not an importable package
```